### PR TITLE
Removed duplicate System.gc() in test

### LINKED ISSUE #18335: Remove duplicate System.gc() in test
Some tests call GC twice.

https://github.com/eclipse-openj9/openj9/blob/fadb895178d06504a9e0ff7d6d8963879cb45a28/test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java#L357..%23L358

This is no longer needed in the tests (see https://github.com/eclipse-openj9/openj9/pull/18044). We should clean it up and call System.gc() once.

FYI @dmitripivkine

### PR DESCRIPTION
**Pull request by flooxo**
Removes that System.gc() is not called repeatedly, but only once in all test files

This PR closes #18335 due to #18044

### CODE REVIEW

**Comment by llxia** [comment]
https://openj9-jenkins.osuosl.org/job/Grinder/2996/

**Comment by hangshao0** [comment]
FYI @theresa-m `ValueTypeTests.java` is being updated here.

File: test/functional/cmdLineTests/gcRegressionTests/src/com/ibm/tests/garbagecollector/TestContractionMain.java [hunk 1]
```diff
@@ -67,12 +67,8 @@ public static void main(String[] args)
 			long highWaterMem = runtime.totalMemory();
 			_array = null;
 			System.gc();
-			System.gc();
-			System.gc();
```
Comment (llxia) [inline test/functional/cmdLineTests/gcRegressionTests/src/com/ibm/tests/garbagecollector/TestContractionMain.java:71]: Talked with @dmitripivkine , 3 GC in the row are needed here. Sorry for misleading. Please rever this change. Same for L74-75.

Comment (flooxo) [reply test/functional/cmdLineTests/gcRegressionTests/src/com/ibm/tests/garbagecollector/TestContractionMain.java]: Okay, thank you. I have reverted the three GC calls again .

**Comment by llxia** [comment]
jenkins test sanity,extended xlinux jdk8,jdk11

**Comment by flooxo** [comment]
I'm not sure right now what to change to make the 3 tests pass. Could you please give me a hint?

**Review by llxia — APPROVED** [review]
LGTM

File: test/functional/JIT_Test/src/jit/test/finalizer/TestObject.java [hunk 2]
```diff
@@ -66,12 +66,10 @@ public static boolean isFinalized(String name) {
 	}
 	
 	protected static void thoroughGCandFinalization() {
-		System.gc();
 		System.gc();
 		System.runFinalization();
```
Comment (pshipton) [inline test/functional/JIT_Test/src/jit/test/finalizer/TestObject.java:70]: @dmitripivkine Do we still need to call runFinaliization() and do the additional System.g()? According to our doc update the finalization should occur from the single System.gc() now.

File: test/functional/VM_Test/src/j9vm/test/classunloading/ClassUnloadingTestParent.java [hunk 3]
```diff
@@ -61,12 +61,10 @@ public void runTest() throws Exception {
 }
 
 protected static void thoroughGCandFinalization() {
-	System.gc();
 	System.gc();
 	System.runFinalization();
```
Comment (pshipton) [inline test/functional/VM_Test/src/j9vm/test/classunloading/ClassUnloadingTestParent.java:65]: Same comment as TestObject.java, do we still need to call runFinaliization() and do the additional System.gc()? According to our doc update the finalization should occur from the single System.gc() now.

File: test/functional/JIT_Test/src/jit/test/finalizer/TestObject.java (hunk 2 above)
Comment (dmitripivkine) [reply test/functional/JIT_Test/src/jit/test/finalizer/TestObject.java:70]: > @dmitripivkine Do we still need to call runFinaliization() and do the additional System.gc()? According to our doc update the finalization should occur from the single System.gc() now.

In theory your statement is correct. However I don't know the reason why we call `System.gc()` and `System.runFinalization()` multiple times. I guess there is a reason. We can try to simplify this code (calling System.gc() twice?) and watch for intermittent failures.

File: test/functional/VM_Test/src/com/ibm/j9/monitor/tests/LockNurseryTest.java [hunk 4]
```diff
@@ -72,8 +72,6 @@ public static void generateGarbageAndCollect(){
 		}
 		System.out.println("Finished generating garbage, requesting gc");
 		System.gc();
-		System.gc();
-		System.gc();
```
Comment (dmitripivkine) [inline test/functional/VM_Test/src/com/ibm/j9/monitor/tests/LockNurseryTest.java:76]: We should watch this case too. I don't know exact reason we called `System.gc()` three times. We do have race condition in Metronome when we can not guarantee Global GC execution even we call it twice.

**Comment by dmitripivkine** [comment]
@pshipton What approach do you prefer? We can update all double and triple SystemGC() calls. However I am almost sure it will introduce number of highly intermittent failures. And we can address these failures by adjusting tests again. Alternatively we can do adjustments more accurate assuming current code is doing necessary operations. So, replace triple SystemGC() call to double, call extra runFinaliization() etc.

**Comment by pshipton** [comment]
We can reduce the calls to what we think is the minimum required now and then monitor the tests. If System.gc() does finalization then do we just need to call System.gc() twice and remove the rest?

**Comment by pshipton** [comment]
Or maybe this is the min for those two places.

System.gc();
System.runFinaliziation();
System.gc();
System.runFinalization();

**Comment by pshipton** [comment]
@flooxo pls make the changes as described in https://github.com/eclipse-openj9/openj9/pull/18339#issuecomment-1787390812

**Comment by flooxo** [comment]
Sure, let me know if it is now as intended or if I understood something wrong :)

File: test/functional/VM_Test/src/com/ibm/j9/monitor/tests/LockNurseryTest.java (hunk 4 above)
Comment (pshipton) [reply test/functional/VM_Test/src/com/ibm/j9/monitor/tests/LockNurseryTest.java:76]: This should retain two calls to System.gc()

**Comment by llxia** [comment]
@pshipton @dmitripivkine could you take a second look? Thanks

**Review by pshipton — APPROVED** [review]

**Review by dmitripivkine — APPROVED** [review]

**Comment by llxia** [comment]
jenkins test sanity,extended xlinux jdk8,jdk11

**Comment by llxia** [comment]
rerun:
https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_1/211/
https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Personal/38/
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.functional_x86-64_linux_Personal_testList_1/54/

**Pull request merged**